packages/react-router/modules/Router.js

```diff
@@ -53,14 +54,18 @@ class Router extends React.Component {
   render() {
     return (
       <RouterContext.Provider
-        children={this.props.children || null}
         value={{
           history: this.props.history,
```

Comment:
I am not quite sure why we should keep `history` in the general router context when we have it also in the history only router.

Reply:
For semver. We can't remove it, as people make be accessing it directly. They shouldn't, but they may still be.

Reply:
I see. So it should be a breaking change, maybe to RRv6.

Reply:
It is worth considering in the design of v6, but for now it would be better to just update the documentation to recommend using the history context.